mathematical-companion/main.tex

```diff
@@ -0,0 +1,1300 @@
+a long time to fill out and provides the user no feedback until the very end, at
+which point the result might just be ``wrong residue, start over''. To avoid this
+frustration, we recommend the following process for derived shares and recovery:
+\begin{enumerate}
+\item Before deriving any shares, complete Checksum Worksheets for the inital
+shares. (During construction these should be readily available; during recovery
+it's worth doing as a sanity check.)
+\item Derive the actual share, using the above method.
+\item Copy the share into the top diagonal, i.e.~the bolded data cells, of a fresh
+Checksum Worksheet.
+\item Derive the cells of the \emph{bottom} diagonal directly, in the same way that
+you derived the top diagonal.
+\item Start working through the worksheet.
+\end{enumerate}
+
+If, at any point, the computed value of a bottom diagonal square doesn't match
+the precomputed value, it means you made a mistake in that column. Re-derive the
+top and bottom values and redo the additions before continuing.
+
+\section{Quickchecks}
+
+\paragraph{Important Note.} {\color{BrickRed}This section describes a series of
+worksheets which, as of March 2023, do not actually exist.}
+
+Recall that the generating polynomial for our primary code is
+\begin{align*}
+    G(x) = x^{13}
+      &+ \vc{E}x^{12} + \vc{M}x^{11} + \vc{3}x^{10} + \vc{G}x^9 + \vc{Q}x^8 + \vc{E}x^7 \\
+      &+ \vc{E}x^6 + \vc{E}x^5 + \vc{L}x^4 + \vc{M}x^3 + \vc{C}x^2 + \vc{S}x + \vc{S}.
+\end{align*}
+Previously we simply took this polynomial as a given. But in this section and the
+next, we need to dig a bit deeper into how it was constructed.
+
+To this end we first construct the extension field $\ftttwo$. Just as we
+produced $\fttwo$ by adding a root of the irreducible 5th-degree polynomial
+$x^5 + x^3 + 1$ over $\ftwo$, we can adjoin a root of the irreducible
+2nd-degree polynomial $x^2 + x + 1$ to $\fttwo$ to get a new field $\ftttwo$.
+We will call this new root $\zeta$. By construction, $\zeta$ satisfies
+the equation $\zeta^2 = \zeta + 1$.
+
+We can write any element of $\ftttwo$ as $a\zeta + b$, where $a$ and $b$ are
+in $\fttwo$. Multiplication and addition happen in the obvious way, with
+every $\zeta^2$ factor simply replaced by $\zeta + 1$.
+
+The new field $\ftttwo$ has 1024 elements; its multiplicative group has
+$1023 = 3\cdot11\cdot31$ elements, so unlike the case of $\fttwo$ where every
+non-unit element has order 31, in $\ftttwo$ non-unit elements might have any
+order in the set $\{3, 11, 31, 33, 93, 341, 1023\}$.
+
+You may recognize the numbers 93 and 1023 as the length of our primary and
+alternate code (and 1023 as the length of bech32). This is not a coincidence.
+
+Consider the element $\beta = \vc{G}\zeta$, which has order $93$. In fact, our
+generator $G$ has roots which are all powers of $\beta$! We can write it as
+\begin{align*}
+    G(x) = \prod_{i\in\{17, 20, 46, 49, 52, 77, 78, 79, 80, 81, 82, 83, 84\}} (x - \beta^i).
+\end{align*}
```

Comment:
A slightly more honest phrasing is that we took the minimal polynomials of \beta^77 … \beta^84 and multiplied them all together.  That does require you to define what a minimal polynomial is thought.

Reply:
I think I already talk about the "degree of a field element", which I may not have defined (it is the degree of the element's minimal polynomial). So I'll make sure there's a definition of minimum polynomial then reword this.